modules/storage/src/service/s3.rs

```diff
@@ -47,11 +60,18 @@ impl StorageBackend for S3Backend {
 
         let stream = pin!(stream);
         let mut file = TempFile::new(stream).await.map_err(errhdlr)?;
-        let mut source = file.reader().await.map_err(errhdlr)?;
+        let mut source = self
+            .compression
+            .compress(file.reader().await.map_err(errhdlr)?)
+            .await;
         let result = file.result();
 
         self.bucket
-            .put_object_stream(&mut source, result.key().to_string())
+            .put_object_stream_with_content_type(
+                &mut source,
+                result.key().to_string(),
+                "application/json",
```

Comment:
Does that require all documents to be JSON? IIRC, that's not the case for some OSV documents, which are YAML instead.

Reply:
Good call. I realized we don't even use the content-type, so the default will do. Besides, I don't think we want to ripple a content-type argument through our existing storage api.